src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -441,6 +444,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     fun addSuperinterface(superinterface: KClass<*>)
         = addSuperinterface(superinterface.asTypeName())
 
+    fun addSuperinterface(superinterface: ParameterSpec) = apply {
```

Comment:
Seems reasonable. Might also want the `String` overload for the parameter name since it's the common case.